Doc/library/imaplib.rst

```diff
@@ -307,6 +308,91 @@ An :class:`IMAP4` instance has the following methods:
    of the IMAP4 QUOTA extension defined in rfc2087.
 
 
+.. method:: IMAP4.idle(dur=None)
+
+   Return an iterable context manager implementing the ``IDLE`` command
+   as defined in :rfc:`2177`.
+
+   The context manager sends the ``IDLE`` command when activated by the
+   :keyword:`with` statement, produces IMAP untagged responses via the
+   :term:`iterator` protocol, and sends ``DONE`` upon context exit.
+
+   The *dur* argument sets a maximum duration (in seconds) to keep idling,
+   after which iteration will stop.  It defaults to ``None``, meaning no time
+   limit.  Callers wishing to avoid inactivity timeouts on servers that impose
+   them should keep this <= 29 minutes.
+   See the :ref:`warning below <windows-pipe-timeout-warning>` if using
+   :class:`IMAP4_stream` on Windows.
+
+   Response tuples produced by the iterator almost exactly match those
+   returned by other imaplib methods.  The difference is that the tuple's
+   second member is a single response datum, rather than a list of data.
+   Therefore, in a mailbox where calling ``M.response('EXISTS')`` would
+   return ``('EXISTS', [b'1'])``, the idle iterator would produce
+   ``('EXISTS', b'1')``.
+
+   Example::
+
+      with M.idle(dur=29*60) as idler:
+          for response in idler:
+              typ, datum = response
+              print(typ, datum)
+
+      ('EXISTS', b'1')
+      ('RECENT', b'1')
+
+   Instead of iterating one response at a time, it is also possible to retrieve
+   the next response along with any immediately available subsequent responses
+   (e.g. a rapid series of ``EXPUNGE`` events from a bulk delete).  This
+   batch processing aid is provided by the context's ``burst()``
+   :term:`generator`:
+
+   .. method:: idler.burst(interval=0.1)
```

Comment:
I think we should rename `_Idler` to `Idler`, because prefixing it with `_` forces compromises like this to exist. `idler` is not something that imaplib has, so this is a little ambiguous. It should be `Idler.burst`, and the `idle` method should document that it returns an `Idler`.